src/components/templates/PartyMap/components/RoomModal/RoomModal.tsx

```diff
@@ -37,13 +39,38 @@ export const RoomModal: React.FC<RoomModalProps> = ({
   room,
   show,
   venue,
-}) => (
-  <Modal show={show} onHide={onHide}>
-    <div className="room-modal">
-      {room && venue && <RoomModalContent room={room} venueName={venue.name} />}
-    </div>
-  </Modal>
-);
+}) => {
+  let template;
+
+  switch (room?.type) {
+    case RoomTypes.video:
+      template = (
+        <VideoModal
+          show={show}
+          onHide={onHide}
+          caption={room.title}
+          url={room.url}
+          autoplay={true}
+          backdrop={true}
+        />
+      );
+      break;
+
+    case RoomTypes.pdf:
+      template = <PdfModal show={show} onHide={onHide} url={room.url} />;
+      break;
+
+    default:
+      template = venue && room && (
+        <Modal show={show} onHide={onHide}>
+          <div className="room-modal">
+            <RoomModalContent room={room} venueName={venue.name} />
+          </div>
+        </Modal>
+      );
+  }
+  return <>{template}</>;
+};
```

Comment:
Probably no need to use `let`, the fragment around `template`, etc here. Looks like you may as well just `return` straight from the `switch`/`case` options?

Reply:
This is now fixed. Thanks